File: FIL.Web.Admin/ClientApp/models/ReportingWizard/ReportResponseDataViewModel.ts
﻿import ReportColumn from "../Common/ReportingColumn";
import TransactionDetail from "../Common/TransactionDetailViewModel";
import TicketCategory from "../Common/TicketCategoryViewModel";
import User from "../Common/UserViewModel";
import TransactionDeliveryDetail from "../Common/TransactionDeliveryDetailViewModel";
import currencyType from "../Common/CurrencyTypeViewModel";
import TransactionPaymentDetail from "../Common/TransactionPaymentDetailViewModel";
import EventTicketDetail from "../Common/EventTicketDetailViewModel";
import EventTicketAttribute from "../Common/EventTicketAttributeViewModel";
import Transaction from "../Common/TransactionViewModel";
import Event from "../Common/EventViewModel";
import EventDetail from "../Common/EventDetailViewModel";
import Venue from "../Common/VenueViewModel";
import City from "../Common/CityViewModel";
import State from "../Common/StateViewModel";
import Country from "../Common/CountryViewModel";
import EventAttribute from "../Common/EventAttributeViewModel";
import CurrencyType from "../Common/CurrencyTypeViewModel";
import IPDetail from "../Common/IPDetailViewModel";
import UserCardDetail from "../Common/UserCardDetailViewModel";
import TicketFeeDetail from "../Common/TicketFeeDetail";

export class ReportResponseDataViewModel {
    transaction: Transaction[];
    transactionDetail: TransactionDetail[];
    transactionDeliveryDetail: TransactionDeliveryDetail[];
    transactionPaymentDetail: TransactionPaymentDetail[];
    currencyType: CurrencyType[];
    eventTicketAttribute: EventTicketAttribute[];
    eventTicketDetail: EventTicketDetail[];
    ticketCategory: TicketCategory[];
    event: Event[];
    eventDetail: EventDetail[];
    eventAttribute: EventAttribute[];
    venue: Venue[];
    city: City[];
    state: State[];
    country: Country[];
    user: User[];
    userCardDetail: UserCardDetail[];
    reportColumns: ReportColumn[];
    ipDetail: IPDetail[];
    ticketFeeDetail: TicketFeeDetail[];
}

File: FIL/FIL.Web.Admin/ViewModels/Reporting/ReportResponseDataViewModel.cs
using FIL.Contracts.Enums;
using FIL.Contracts.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace FIL.Web.Kitms.Feel.ViewModels.Reporting
{
    public class ReportResponseDataViewModel
    {
        public IEnumerable<FIL.Contracts.Models.Transaction> Transaction { get; set; }
        public IEnumerable<TransactionDetail> TransactionDetail { get; set; }
        public IEnumerable<TransactionDeliveryDetail> TransactionDeliveryDetail { get; set; }
        public IEnumerable<TransactionPaymentDetail> TransactionPaymentDetail { get; set; }
        public IEnumerable<CurrencyType> CurrencyType { get; set; }
        public IEnumerable<EventTicketAttribute> EventTicketAttribute { get; set; }
        public IEnumerable<EventTicketDetail> EventTicketDetail { get; set; }
        public IEnumerable<TicketCategory> TicketCategory { get; set; }
        public IEnumerable<FIL.Contracts.Models.Event> Event { get; set; }
        public IEnumerable<EventDetail> EventDetail { get; set; }
        public IEnumerable<EventAttribute> EventAttribute { get; set; }
        public IEnumerable<FIL.Contracts.Models.Venue> Venue { get; set; }
        public IEnumerable<FIL.Contracts.Models.City> City { get; set; }
        public IEnumerable<FIL.Contracts.Models.State> State { get; set; }
        public IEnumerable<FIL.Contracts.Models.Country> Country { get; set; }
        public IEnumerable<FIL.Contracts.Models.User> User { get; set; }
        public IEnumerable<FIL.Contracts.Models.UserCardDetail> UserCardDetail { get; set; }
        public IEnumerable<FIL.Contracts.Models.ReportingColumn> ReportColumns { get; set; }
        public IEnumerable<FIL.Contracts.Models.IPDetail> IPDetail { get; set; }
        public IEnumerable<TicketFeeDetail> TicketFeeDetail { get; set; }
        public  Pagination Pagination { get; set; }
    }
}
